adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -88,15 +88,30 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts []byte,
 	conf, _ := config.ReadConfigs()
 	timeout := conf.Envoy.ClusterTimeoutInSeconds
 
+	// check if x-wso2-endpoints are available
+	xWso2Endpoints := mgwSwagger.GetXWso2Endpoints()
+	if len(xWso2Endpoints) > 0 {
+		logger.LoggerOasparser.Debug("x-wso2-endpoints clusters found")
+		for _, endpointCluster := range xWso2Endpoints {
+			epClusterName := strings.TrimSpace(organizationID + "_" + endpointCluster.EndpointPrefix + "_" + vHost +
+				"_" + strings.Replace(mgwSwagger.GetTitle(), " ", "", -1) + mgwSwagger.GetVersion())
+			epCluster, addresses := createCluster(epClusterName, endpointCluster, upstreamCerts, timeout)
+			clusters = append(clusters, epCluster)
+			endpoints = append(endpoints, addresses...)
+		}
+	}
+
 	// check API level production endpoints available
 	if mgwSwagger.GetProdEndpoints() != nil && len(mgwSwagger.GetProdEndpoints().Endpoints) > 0 {
 		apiLevelEndpointProd = mgwSwagger.GetProdEndpoints()
-		apiLevelClusterNameProd = strings.TrimSpace(organizationID + "_" + prodClustersConfigNamePrefix + vHost + "_" +
+		apiLevelClusterNameProd = strings.TrimSpace(organizationID + "_" + apiLevelEndpointProd.EndpointPrefix + "_" + vHost + "_" +
 			strings.Replace(mgwSwagger.GetTitle(), " ", "", -1) + mgwSwagger.GetVersion())
```

Comment:
what if we have one place to generate cluster name, with the expected prefix

Reply:
fixed